Python log injection traceID/spanID (#1408)
SumoLogic/sumologic-documentation: docs/apm/traces/get-started-transaction-tracing/opentelemetry-instrumentation/python/traceid-spanid-injection-into-logs.md

---
id: traceid-spanid-injection-into-logs
title: Python TraceId and SpanId injection into logs configuration
sidebar_label: TraceId and SpanId injection into logs
description: Learn how to configure traceId, spanId, and trace flags data injection into user logs in Python applications.
---

> Here's how to add traceId and spanId into logs for Python lambda. You have to get a context from current span and attach it to the log.

Author: ```suggestion
Here's how to add `trace_id` and `span_id` into your Python lambda logs. You have to get a context from current span and attach it to the log.

:::important
Make sure that the code from below is active in the `.py` file where your application requests are traced with OpenTelemetry python instrumentation. Logs not related to traced requests will not have `span_id` or `trace_id` filled in.
:::
```

Author: @pbrzoska-sumo , does this capture your change correctly? ☝️

## Python lambda

1. Add library import:
  ```py

>   from OpenTelemetry import trace

Reviewer: no capitals in the code please

>   from OpenTelemetry import trace

Author: ```suggestion
  from opentelemetry import trace
```

---
id: traceid-spanid-injection-into-logs
title: Python TraceId and SpanId injection into logs configuration
sidebar_label: TraceId and SpanId injection into logs
description: Learn how to configure traceId, spanId, and trace flags data injection into user logs in Python applications.
---

> Here's how to add `trace_id` and `span_id` into your Python lambda logs. You have to get a context from current span and attach it to the log.

Author: ```suggestion
To add `trace_id` and `span_id` to your Python lambda logs, you'll have to get a context from current span and attach it to the log.
```
